src/t_zset.c

```diff
@@ -72,12 +72,41 @@ void zsetConvertAndExpand(robj *zobj, int encoding, unsigned long cap);
 zskiplistNode *zslGetElementByRankFromNode(zskiplistNode *start_node, int start_level, unsigned long rank);
 zskiplistNode *zslGetElementByRank(zskiplist *zsl, unsigned long rank);
 
+static inline unsigned long zslGetNodeSpanAtHeight(zskiplistNode *x, int height) {
+    return height > 0 ? x->level[height].span : x->level[height].forward ? 1
+                                                                         : 0;
+}
+
+static inline void zslSetNodeSpanAtHeight(zskiplistNode *x, int height, unsigned long span) {
+    if (height > 0)
+        x->level[height].span = span;
+}
+
+static inline void zslIncrNodeSpanAtHeight(zskiplistNode *x, int height, unsigned long incr) {
+    if (height > 0)
+        x->level[height].span += incr;
+}
+
+static inline void zslDecrNodeSpanAtHeight(zskiplistNode *x, int height, unsigned long decr) {
+    if (height > 0)
+        x->level[height].span -= decr;
+}
+
+static inline unsigned long zslGetNodeHeight(zskiplistNode *x) {
+    return x->level[0].span;
```

Comment:
The span of level 0 used for storing the height? This is unintuitive so let's add a comment for this, here and there. This code looks like a typo if you don't know this encoding.

```suggestion
    /* Since the span at level 0 is always 1 (or 0 for the last node), this
     * field is instead used for storing the height of the node. */
    return x->level[0].span;
```

Reply:
I added different comments to explain the change